src/Go.CodeAnalysis/Text/SnapshotSegment.cs

```diff
@@ -29,14 +32,32 @@ public SnapshotSegment(SnapshotBase snapshotBase, int start, int length)
 
         public char this[int offset] => this.Snapshot[this.Start + offset];
 
+        /// <summary>
+        /// Starting position of the code segment.
+        /// </summary>
         public int Start { get; }
 
+        /// <summary>
+        /// Length of the code segment.
+        /// </summary>
         public int Length { get; }
 
+        /// <summary>
+        /// Ending position of the code segment.
```

Comment:
Worth noting that this is an ‘open ended span’, meaning that the End point is always _after_ the last character in the span.